enhancements/automated-service-ca-rotation.md

```diff
@@ -99,6 +99,31 @@ and the functionality could be separately extracted for operator reuse.
   - This provides a 6 month grace period for consumers of serving certs and ca
     bundles to be refreshed.
 
+- The duration of the service CA should be extended from its current value of
+  12 months to a value that exceeds the maximum expected upgrade interval
+  (currently 12 months). A value of 14 months should be sufficient.
+  - Timelines:
+    - With 12 month CA duration:
+      - T+0m  - Cluster installed with new CA or existing CA is rotated (CA-1)
+      - T+6m  - Automated rotation replaces CA-1 with CA-2 when CA-1 duration < 6m
+      - T+12m - Cluster is upgraded and all pods are restarted
+      - T+12m - CA-1 expires. If cluster was not upgraded before this
+                happens, services using the old key material may
+                break.
+    - With 14 month CA duration:
+      - T+0m  - Cluster installed with new CA or existing CA is rotated (CA-1)
+      - T+8m  - Automated rotation replaces CA-1 with CA-2 when CA-1 duration < 6m
+      - T+12m - Cluster is upgraded and all pods are restarted
+      - T+14m - CA-1 expires. No impact because of the restart at time of upgrade
+  - Services that do not refresh key material automatically must be restarted
+    after ca rotation.
+  - Upgrades restart all pods, and ensuring that an upgrade takes place after
```

Comment:
this happens because nodes are upgraded, which involves a restart and therefore a remount of configmaps?

Reply:
Correct. And automated rotation at time of upgrade is not guaranteed to occur prior to the node upgrade, post-rotation restart must rely on a subsequent upgrade (or manual restart).